lib/ansible/plugins/action/uri.py

```diff
@@ -20,6 +20,8 @@ class ActionModule(ActionBase):
     TRANSFERS_FILES = True
 
     def run(self, tmp=None, task_vars=None):
+        self._supports_async = True
+
```

Comment:
this is not only thing you need, you also need to ensure you don't remove temp dirs as that is now the job of async_wrapper

Reply:
you also need to tell execute module to 'wrap async'

Reply:
I'll go check more.  I honestly wasn't aware this was necessary until today.

Reply:
Ok, all of the remaining needed changes have been made.